### PR TITLE
CatBoost recipe for conda-forge

### PR DESCRIPTION
**Pull request by sab**

### CODE REVIEW

**Comment by sab** [comment]
This is recipe for machine learning library CatBoost (https://github.com/catboost/catboost). 

@conda-forge/staged-recipes ping. Does anything else need to add into this PR?

File: recipes/catboost/build.sh [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+
+# install using pip from the whl files on PyPI
+
+if [ `uname` == Darwin ]; then
+    if [ "$PY_VER" == "2.7" ]; then
+        WHL_FILE=https://pypi.org/packages/cp27/c/catboost/catboost-${PKG_VERSION}-cp27-none-macosx_10_6_intel.macosx_10_9_intel.macosx_10_9_x86_64.macosx_10_10_intel.macosx_10_10_x86_64.whl
+    elif [ "$PY_VER" == "3.4" ]; then
+        WHL_FILE=https://pypi.org/packages/cp34/c/catboost/catboost-${PKG_VERSION}-cp34-none-macosx_10_6_intel.macosx_10_9_intel.macosx_10_9_x86_64.macosx_10_10_intel.macosx_10_10_x86_64.whl
+    elif [ "$PY_VER" == "3.5" ]; then
+        #WHL_FILE=https://pypi.org/packages/cp35/c/catboost/catboost-${PKG_VERSION}-cp35-none-macosx_10_6_intel.macosx_10_9_intel.macosx_10_9_x86_64.macosx_10_10_intel.macosx_10_10_x86_64.whl
```
Comment (aldanor) [inline recipes/catboost/build.sh]: Why is this one not from pypi?

Comment (sab) [reply recipes/catboost/build.sh]: It's fixed version for python 3.5. conda-forge CI build fails with CatBoost 0.6.1/2 now. 
Will be changed to proper pypi with next release (planned on the next week).

**Comment by sab** [comment]
Now it looks good.

@CJ-Wright could you or somebody from conda-forge team take a look on this recipe? Thank you!

**Comment by aldanor** [comment]
I've checked, the 0.6.3 wheel for 3.6 on Linux doesn't seem right: https://github.com/catboost/catboost/issues/268, could you take a look?

**Comment by sab** [comment]
@aldanor As you may see we've fixed https://github.com/catboost/catboost/issues/268 issue.
What do you think? Something else prevents us from adding to conda-forge?

**Comment by aldanor** [comment]
Looks good to me; I've checked, the issue seems to be gone indeed in the latest wheel.

@CJ-Wright @ocefpaf @scopatz could this get merged please? Thanks!

**Comment by CJ-Wright** [comment]
Was there any particular reason to not build from source?

**Comment by scopatz** [comment]
Yeah, I agree with @CJ-Wright - I am not sure this is the best way to handle compiled libraries.  I would be very concerned that this is not ABI compatible with other conda-forge packages and may not successfully link.

**Comment by kizill** [comment]
Main problem is our custom build system - ymake.
Another possible problem is building CUDA 8.0 compatible part of python wheel. Currently it compiles for about 2 hours on single threaded build agents. And also we are planning to switch CUDA version to 9.X.

**Comment by kizill** [comment]
We have travis-ci (https://travis-ci.org/catboost/catboost/) for linux and appveyor (https://ci.appveyor.com/project/sab/catboost ) for windows. And yes, we are building mac os wheels on our macbooks :)

**Comment by sodre** [comment]
I second @scopatz concern with ABI compatibility. Also, I think the community is better off building each dependency separately. For example, if ymake is a build-time dependency then make it a conda package. Nowadays, the cost of keeping them up-to-date is small thanks to the auto-tick bot.

**Comment by scopatz** [comment]
For OSX, what SDK are you targeting? IIRC the rest of conda-forge is on 10.9

**Comment by exprmntr** [comment]
@CJ-Wright @scopatz
> Was there any particular reason to not build from source?

Reasons are:
1. Too large package size (about 300M of sources)
2. Too slow compilation (about an hour on a laptop)
3. Installation will depend on compilers and build tools:
     * It can use system build tools and compilers (so installation becomes non-reproducible)
     * It can download build tools and compilers from our hosts (so it will be not possible to install package from file on a machine without internet connection)
     * Build tools and compilers can be distributed inside this packages or with separate packages (so distribution will be about several gigabytes total size)

We realize possible problems with API compatibility. I think, our task in this case - provide correctly built binaries.

**Comment by scopatz** [comment]
I guess you are going to have the same problems with ABI compatability that you do with the wheels, which don't sound super robust as it is.  I am OK with merging this with the understanding that this is really not the correct way to have guaranteed conda-forge compatible packages. I'll merge in 24 hrs unless there are objections.

**Pull request merged**